packages/app/src/cli/services/dev/extension/utilities.ts

```diff
@@ -1,22 +1,23 @@
 import {UIExtension} from '../../../models/app/extensions.js'
+
 import {fetchProductVariant} from '../../../utilities/extensions/fetch-product-variant.js'
 
 /**
- * To prepare Checkout UI Extensions for dev'ing we need to retrieve a valid product variant ID
+ * To prepare UI Extensions targeting Checkout for dev'ing we need to retrieve a valid product variant ID
  * @param extensions - The UI Extensions to dev
  * @param store - The store FQDN
  */
 export async function getCartPathFromExtensions(extensions: UIExtension[], store: string, checkoutCartUrl?: string) {
-  const hasUIExtension = extensions.filter(getExtensionNeedsCartURL).length > 0
+  const hasUIExtension = extensions.filter((extension) => extension.shouldFetchCartUrl()).length > 0
   if (!hasUIExtension) return undefined
   if (checkoutCartUrl) return checkoutCartUrl
   const variantId = await fetchProductVariant(store)
   return `/cart/${variantId}:1`
 }
 
 /**
- * Returns true if an extension needs a cart URL.
+ * Returns the surface for UI extension from an extension point target
  */
-export function getExtensionNeedsCartURL(extension: UIExtension) {
-  return extension.configuration.type === 'checkout_ui_extension'
+export function getExtensionPointTargetSurface(extensionPointTarget: string) {
+  return extensionPointTarget.toLowerCase().replace(/::.+$/, '')
```

Comment:
For now I guess this would work in all cases, but is it possible in the future to have different URLs for the same surface? (maybe taking you a different place in admin)